src/twisted/trial/runner.py

```diff
@@ -724,8 +724,27 @@ def findByName(self, _name, recurse=False):
                 break
 
             except ImportError:
+                # Check to see where the ImportError happened. If it happened
+                # in this file, ignore it.
+                tb = sys.exc_info()[2]
+
+                # Walk down to the deepest frame, where it actually happened.
+                while tb.tb_next is not None:
+                    tb = tb.tb_next
+
+                # Get the filename that the ImportError originated in.
+                filenameWhereHappened = tb.tb_frame.f_code.co_filename
+
+                # If it originated in the reflect file, then it's because it
+                # doesn't exist. If it originates elsewhere, it's because an
+                # ImportError happened in a module that does exist.
+                if filenameWhereHappened != reflect.__file__:
+                    raise
+
                 if remaining == "":
```

Comment:
this can never be true. `remaining` is assigned in two places: [to `None` in line 710](https://github.com/twisted/twisted/pull/1143/files#diff-c8eebbb513e06cd3494e0e38f6536198R710) and [to `remainingName` (the class, method, function etc. part of the fqpn) on line 723](https://github.com/twisted/twisted/pull/1143/files#diff-c8eebbb513e06cd3494e0e38f6536198R723).  in the second case it will be a string, but: a) the `break` prevents this block from running after that assignment, and b) `remaining` can only be `""` with an fqpn that ends in a dot, which is invalid.

the `ModuleNotFound` exception caught by the test comes from [`namedAny` on line 752](https://github.com/twisted/twisted/pull/1143/files#diff-c8eebbb513e06cd3494e0e38f6536198R752).  this should be changed to `if remaining is None:` for it to have the intended effect.